src/Integration.Vsix.UnitTests/SonarLintDaemon/SonarLintDaemonTests.cs

```diff
@@ -58,24 +58,86 @@ public void SetUp()
         [TestCleanup]
         public void CleanUp()
         {
+            testableDaemon.Dispose();
             ForceDeleteDirectory(tempPath);
             ForceDeleteDirectory(storagePath);
         }
 
+        [TestMethod]
+        public void IsRunning_NotRunningOnCreation()
+        {
+            testableDaemon.IsRunning.Should().BeFalse();
+        }
 
         [TestMethod]
-        public void Not_Installed()
+        public void IsInstalled()
         {
+            // Sanity check - not expecting the exe directory to exist to start with
+            var exeDirectory = Path.GetDirectoryName(testableDaemon.ExePath);
+            Directory.Exists(exeDirectory).Should().BeFalse();
+
+            // 1. No directory or file -> false
             testableDaemon.IsInstalled.Should().BeFalse();
-            testableDaemon.IsRunning.Should().BeFalse();
+
+            // 2. Directory and file exist -> true
+            Directory.CreateDirectory(exeDirectory);
+            File.WriteAllText(testableDaemon.ExePath, "junk");
+            testableDaemon.IsInstalled.Should().BeTrue();
         }
 
         [TestMethod]
         public void Run_Without_Install()
         {
             Action op = () => testableDaemon.Start();
 
-            op.Should().ThrowExactly<InvalidOperationException>();
+            op.Should().ThrowExactly<InvalidOperationException>().And.Message.Should().Be("Daemon is not installed");
+        }
+
+        [TestMethod]
+        public void Run_With_Install_Succeeds()
+        {
+            testableDaemon.Port.Should().Be(0);
+
+            // Fake the installation i.e. create the directory and file to execute.
+            testableDaemon.SetUpDummyInstallation("test1.bat",
+@"@echo Hello world
+@echo write to error stream... 1>&2
+");
+
+            // Act
+            testableDaemon.Start();
+            bool processFinished = testableDaemon.process.WaitForExit(1000);
+            processFinished.Should().BeTrue("Test execution error: timed out waiting for the dummy process to exit");
```

Comment:
strictly speaking, line 110 should be after `//Asert` comment

Reply:
For me, line 110 is a sanity check on the execution of the test, and I put those as close to the relevant bit of test code as possible.
I don't think of them in the same was as the checks in the `// Assert` section, which are checks on the product code.